lms/djangoapps/instructor/enrollment.py

```diff
@@ -504,7 +521,7 @@ def send_mail_to_student(student, param_dict, language=None):
 
     message_class = ace_emails_dict[message_type]
     message = message_class().personalize(
-        recipient=Recipient(lms_user_id=0, email_address=student),
+        recipient=Recipient(lms_user_id=lms_user_id, email_address=student),
```

Comment:
When I first read this PR, I was very confused by the concept "Make a best effort in every case." Shouldn't the filter or find method return the id when it is there, with no required "best effort"? Now what I understand, after reading this line, is that the required lms_user_id value was not passed to the actual call to the "messenger." This fix does so, and replaces the placeholder 0 there, correct?

Otherwise, looks good to me, If what I understand is correct?

Reply:
In most cases, an LMS user ID exists for the email address to which the email is being sent - but not all cases. For example, there's a feature in the instructor dashboard where an instructor can enroll a user in a course by email address *before* they have an LMS account. In these cases where no LMS account exists, an ID of `0` will still be sent to ACE.

Reply:
Did the above reply answer your question here, @connorhaugh ?